components/tikv_util/src/lib.rs

```diff
@@ -435,31 +435,32 @@ pub fn set_panic_hook(panic_abort: bool, data_dir: &str) {
         .unwrap();
 
     let data_dir = data_dir.to_string();
-    let orig_hook = panic::take_hook();
+
     panic::set_hook(Box::new(move |info: &panic::PanicInfo<'_>| {
-        use slog::Drain;
-        if slog_global::borrow_global().is_enabled(::slog::Level::Error) {
-            let msg = match info.payload().downcast_ref::<&'static str>() {
-                Some(s) => *s,
-                None => match info.payload().downcast_ref::<String>() {
-                    Some(s) => &s[..],
-                    None => "Box<Any>",
-                },
-            };
-            let thread = thread::current();
-            let name = thread.name().unwrap_or("<unnamed>");
-            let loc = info
-                .location()
-                .map(|l| format!("{}:{}", l.file(), l.line()));
-            let bt = backtrace::Backtrace::new();
-            crit!("{}", msg;
-                "thread_name" => name,
-                "location" => loc.unwrap_or_else(|| "<unknown>".to_owned()),
-                "backtrace" => format_args!("{:?}", bt),
-            );
-        } else {
-            orig_hook(info);
-        }
+        let msg = match info.payload().downcast_ref::<&'static str>() {
+            Some(s) => *s,
+            None => match info.payload().downcast_ref::<String>() {
+                Some(s) => &s[..],
+                None => "Box<Any>",
+            },
+        };
+
+        let thread = thread::current();
+        let name = thread.name().unwrap_or("<unnamed>");
+        let loc = info
+            .location()
+            .map(|l| format!("{}:{}", l.file(), l.line()));
+        let bt = backtrace::Backtrace::new();
+        crit!("{}", msg;
+            "thread_name" => name,
+            "location" => loc.unwrap_or_else(|| "<unknown>".to_owned()),
+            "backtrace" => format_args!("{:?}", bt),
+        );
+        // As a convenience show the panic message at the end of the output
+        // This avoids having to look above a long stack trace
+        crit!("{}", msg);
```

Comment:
People may feel confused as the message is output twice. On the other hand, when using slog, backtrace is put in the same line as the message, so the message has to be seen no matter using less, tail or log collectors.

Reply:
I don't think seeing the message twice is confusing as to what happened to TiKV. Quite the opposite. I agree that someone might assume it is a minor bug that it shows up twice, but I don't see why this poses a problem.

> backtrace is put in the same line as the message, so the message has to be seen no matter using less, tail or log collectors
This is true if one views the output in vim for example and I am sure in many log collectors.
However, if one uses less, tail, or otherwise views the terminal output, the backtrace takes up an entire screen and the actual panic message cannot be seen.

One solution to avoiding printing the panic message twice would be to print out the backtrace by itself without the panic message.

Reply:
> ...but I don't see why this poses a problem.

You will when you need to handle customer's issue and being asked the same question again and again.

> ...takes up an entire screen...

Unless it's a very old screen that can only show 80 characters in a line, I don't see how backtrace can take up the whole screnn...And if it did, why not just scrolled up?

> ...print out the backtrace by itself without the panic message.

This is worse. Backtrace and panic message are put at the same line on purpose to reduce communication roundtrip during supporting.

Reply:
> Unless it's a very old screen that can only show 80 characters in a line,  I don't see how backtrace can take up the whole screnn

I feel like you are making statements without bother to take the time looking at the panic output? It is now trivial to look at the panic output according to the instructions in this PR.
At a 134 character width while using an external monitor the panic output takes up a full screen for me (about 2 full vertical screens).

Reply:
Okay @BusyJay , I have one last attempt at convenience: additionally printing the panic message in a slog field "panic". This way it shows up at the very bottom, but it is still all one slog logging statement. This is pushed now.

Reply:
We don't know why, but I see 78 lines of stack traces, whereas @BusyJay normally sees 25.
@BusyJay does not want to make my use case convenient because he is afraid users will not send him a full screenshot of the last log message that includes the backtrace if they can see the last line of the log message already.

Of course the solution to this is to have an optional setting, but @BusyJay is against this as well.

Reply:
The difference in stack trace seems to be compiling debug `make dev` versus release `make release`.